nibabel/processing.py

```diff
@@ -310,3 +311,80 @@ def smooth_image(img,
+
+    Using the default arguments, this function is meant to replicate most parts
+    of FreeSurfer's ``mri_convert --conform`` command. Specifically, this
+    function:
+        - Resamples data to ``output_shape``
+        - Resamples voxel sizes to ``voxel_size``
+        - Reorients to RAS (``mri_convert --conform`` reorients to LIA)
+
+    Unlike ``mri_convert --conform``, this command does not:
+        - Transform data to range [0, 255]
+        - Cast to unsigned eight-bit integer
+
+    Parameters
+    ----------
+    from_img : object
+        Object having attributes ``dataobj``, ``affine``, ``header`` and
+        ``shape``. If `out_class` is not None, ``img.__class__`` should be able
+        to construct an image from data, affine and header.
+    out_shape : sequence, optional
+        The shape of the output volume. Default is (256, 256, 256).
+    voxel_size : sequence, optional
+        The size in millimeters of the voxels in the resampled output. Default
+        is 1mm isotropic.
+    order : int, optional
+        The order of the spline interpolation, default is 3.  The order has to
+        be in the range 0-5 (see ``scipy.ndimage.affine_transform``)
+    cval : scalar, optional
+        Value used for points outside the boundaries of the input if
+        ``mode='constant'``. Default is 0.0 (see
+        ``scipy.ndimage.affine_transform``)
+    orientation : str, optional
+        Orientation of output image. Default is "RAS".
+    out_class : None or SpatialImage class, optional
+        Class of output image.  If None, use ``from_img.__class__``.
+
+    Returns
+    -------
+    out_img : object
+        Image of instance specified by `out_class`, containing data output from
+        resampling `from_img` into axes aligned to the output space of
+        ``from_img.affine``
+    """
+    # Only support 3D images. This can be made more general in the future, once tests
+    # are written.
+    required_ndim = 3
+    if from_img.ndim != required_ndim:
+        raise ValueError("Only 3D images are supported.")
+    elif len(out_shape) != required_ndim:
+        raise ValueError("`out_shape` must have {} values".format(required_ndim))
+    elif len(voxel_size) != required_ndim:
+        raise ValueError("`voxel_size` must have {} values".format(required_ndim))
+
+    # Create fake image of the image we want to resample to.
+    hdr = Nifti1Header()
+    hdr.set_data_shape(out_shape)
+    hdr.set_zooms(voxel_size)
+    dst_aff = hdr.get_best_affine()
+    to_img = Nifti1Image(np.empty(out_shape), affine=dst_aff, header=hdr)
+
+    # Resample input image.
+    out_img = resample_from_to(
+        from_img=from_img, to_vox_map=to_img, order=order, mode="constant",
+        cval=cval, out_class=out_class)
+
+    # Reorient to desired orientation.
+    ornt = axcodes2ornt(orientation, labels=list(zip('RPI', 'LAS')))
+    return out_img.as_reoriented(ornt)
```

Comment:
These `labels` are not the default. I included them because if I created an orientation array for `"RAS"` with the default, the image would be in `"LAS"` orientation. Any idea why this might be?

Reply:
This doesn't seem to take into account your original orientation, so very likely what's happening is that you have an LAS image. `axcodes2ornt("RAS")` will get you an identity transform.

To get the transform to RAS, use [io_orientation(out_img.affine)](https://nipy.org/nibabel/reference/nibabel.orientations.html#nibabel.orientations.io_orientation). To get the target transform, use `axcodes2ornt`.

Example:

```Python
def flip_img(img, axcodes):
    start_ornt = nb.io_orientation(img.affine)
    end_ornt = nb.orientations.axcodes2ornt(axcodes)
    transform = nb.orientations.ornt_transform(to_ras, to_target)
    return img.as_reoriented(transform)
```

Reply:
thank you so much. i've correct the orientation code in `conform`